packages/eth2.0-types/src/types/validator.ts

```diff
@@ -26,3 +27,61 @@ export interface SyncingStatus {
   // The estimated highest block, or current target block number
   highestBlock: uint64;
 }
+
+export interface Registry<T, T1> extends Array<T> {
+  // find index in array from a field
+  findIndexByRegistry?: (field: T1) => number;
+}
+// T: the type to index
+// T1: the type of field to index
+interface InternalRegistry<T, T1> extends Registry<T, T1> {
+  registry?: Map<T1, number>;
+}
+
+export function getRegistry<T, T1>(items: T[], getField: (item: T) => T1): Registry<T, T1> {
+  const internalRegistry: InternalRegistry<T, T1> = items;
+  // index for the first time
+  const tmp = new Map<T1, number>();
+  internalRegistry.forEach((value: T, index: number) => tmp.set(getField(value), index));
+  internalRegistry.registry = tmp;
+  internalRegistry.findIndexByRegistry = (field: T1) => internalRegistry.registry.get(field);
+
+  // on the fly index
+  const handler: ProxyHandler<InternalRegistry<T, T1>> = {
+    get: (target: InternalRegistry<T, T1>, p: PropertyKey) => {
```

Comment:
do you need to define `get`?

Reply:
I don't want to allow accessing `registry` as it's our internal index data.

Reply:
if you don't define it in interface you can just add private modifier